vllm/entrypoints/openai/tool_parsers/internlm2_tool_parser.py

```diff
@@ -0,0 +1,208 @@
+import json
+from typing import Dict, Sequence, Union
+
+import partial_json_parser
+from partial_json_parser.core.options import Allow
+
+from vllm.entrypoints.openai.protocol import (ChatCompletionRequest,
+                                              DeltaFunctionCall, DeltaMessage,
+                                              DeltaToolCall,
+                                              ExtractedToolCallInformation,
+                                              FunctionCall, ToolCall)
+from vllm.entrypoints.openai.tool_parsers.abstract_tool_parser import (
+    ToolParser, ToolParserManager)
+from vllm.entrypoints.openai.tool_parsers.utils import (
+    extract_intermediate_diff)
+from vllm.logger import init_logger
+from vllm.transformers_utils.tokenizer import AnyTokenizer
+from vllm.utils import random_uuid
+
+logger = init_logger(__name__)
+
+
+@ToolParserManager.register_module(["internlm2", "internlm2_5"])
```

Comment:
are there two aliases for the same tool parser? If so, it's probably better to simplify to just 1 - maybe `internlm` or similar. e.g. `hermes` supports hermes 2 pro, hermes 2 theta, and hermes 3 models; as well as qwen 2.5 since they adopted the Hermes tool calling format.

Reply:
and `mistral` supports several different mistral models

Reply:
that's a good idea, i'll change it.